### PR TITLE
add database connection

### LINKED ISSUE #17: Create database connection to mysql database

### PR DESCRIPTION
**Pull request by Flanker-shyam**
resolve #17

### CODE REVIEW

**Comment by Flanker-shyam** [comment]
@debasishbsws , take a look at the code and review !!

File: src/database/connectDB.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const mysql = require('mysql');
+require('dotenv').config();
```
Comment (debasishbsws) [inline src/database/connectDB.ts:2]:
```suggestion
import mysql, {Connection, MysqlError} from 'mysql';
```

File: src/database/connectDB.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+const UserName = process.env.USER_NAME;
+const Password = process.env.PASSWORD;
+const Database = process.env.DATABASE;
+const Host = process.env.HOST;
```
Comment (debasishbsws) [inline src/database/connectDB.ts:7]:
```suggestion
const UserName = process.env.DB_USER_NAME;
const Password = process.env.DB_PASSWORD;
const Database = process.env.DATABASE_NAME;
const Host = process.env.DB_HOST;
```

File: src/database/connectDB.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+let connection = mysql.createConnection({
+    host: Host,
+    user: UserName,
+    password: Password,
+    database: Database
+});
```
Comment (debasishbsws) [inline src/database/connectDB.ts:14]:
```suggestion
const connection: Connection = mysql.createConnection( {
    host: Host as string,
    user: UserName as string,
    password: Password as string,
    database: Database as string
});
```
change it to const as it never reassigning. Add type to the variable

File: src/database/connectDB.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+function openConnection()
+{
+    connection.connect((err: any)=>{
+        if(err)
+        {
+            console.error("Could not connect to the database", err.message);
```
Comment (debasishbsws) [inline src/database/connectDB.ts:21]:
```suggestion
    connection.connect((err: MysqlError)=>{
        if(err)
        {
            console.error("Could not connect to the database", err);
```
add proper types

File: src/database/connectDB.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            console.log("connection to Database is successful.");
+        }
+    })
+}
+
+function closeConnection()
+{
+    connection.end(function(err:any) {
+        if (err) {
+          return console.log('Could not close connection:' + err.message);
```
Comment (debasishbsws) [inline src/database/connectDB.ts:33]:
```suggestion
    connection.end((err?: MysqlError) => {
        if (err) {
          return console.log('Could not close connection:', err);
```

File: src/server.ts
Comment (debasishbsws) [inline src/server.ts:1]: use
`import "dotenv/config";` at the top to use dotenv in the whole project

File: src/database/connectDB.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        }
+        console.log('Connection to the database is closed.');
+    });    
+}
+
+module.exports = {openConnection, closeConnection};
```
Comment (debasishbsws) [inline src/database/connectDB.ts:39]:
```suggestion
export {openConnection, closeConnection};
```
this is the correct syntax for typescript

File: src/database/connectDB.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+const Host = process.env.HOST;
+
+let connection = mysql.createConnection({
+    host: Host,
+    user: UserName,
+    password: Password,
+    database: Database
+});
+
+function openConnection()
```
Comment (debasishbsws) [inline src/database/connectDB.ts:16]:
```suggestion
function openConnection(): void
```
should have a return type

File: src/database/connectDB.ts [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+        {
+            console.error("Could not connect to the database", err.message);
+        }
+        else{
+            console.log("connection to Database is successful.");
+        }
+    })
+}
+
+function closeConnection()
```
Comment (debasishbsws) [inline src/database/connectDB.ts:29]:
```suggestion
function closeConnection(): void
```

**Review by debasishbsws — APPROVED** [review]

**Review by debasishbsws — CHANGES_REQUESTED** [review]
changes required for Typescript lint

**Comment by Flanker-shyam** [comment]
Good to merge now !

**Comment by Flanker-shyam** [comment]
> changes required for Typescript lint

Fixed !

File: src/database/connectDB.ts [hunk 9]
```diff
@@ -1,39 +1,39 @@
-const mysql = require('mysql');
+import mysql,{MysqlError,Connection} from "mysql"
 require('dotenv').config();
```
Comment (debasishbsws) [inline src/database/connectDB.ts:2]:
```suggestion
```
No need for this line here

File: src/database/connectDB.ts [hunk 10]
```diff
@@ -1,39 +1,39 @@
-const mysql = require('mysql');
+import mysql,{MysqlError,Connection} from "mysql"
```
Comment (debasishbsws) [inline src/database/connectDB.ts:1]:
```suggestion
import mysql, {MysqlError, Connection} from "mysql"
```

File: src/database/connectDB.ts [hunk 11]
```diff
@@ -1,39 +1,39 @@
...
-    database: Database
+const connection:Connection = mysql.createConnection({
+    host: Host as string,
+    user: UserName as string,
+    password: Password as string,
+    database: Database as string
 });
 
-function openConnection()
+function openConnection():void
```
Comment (debasishbsws) [inline src/database/connectDB.ts:16]:
```suggestion
function openConnection(): void
```

File: src/database/connectDB.ts [hunk 12]
```diff
@@ -1,39 +1,39 @@
...
         }
         else{
             console.log("connection to Database is successful.");
         }
-    })
+    });
 }
 
-function closeConnection()
+function closeConnection():void
```
Comment (debasishbsws) [inline src/database/connectDB.ts:29]:
```suggestion
function closeConnection(): void
```

**Review by debasishbsws — CHANGES_REQUESTED** [review]
Need some formatting

File: src/app/app.ts [hunk 13]
```diff
@@ -1,9 +1,16 @@
 import express, { Request, Response } from 'express';
+const {openConnection, closeConnection} = require('../database/connectDB');
```
Comment (debasishbsws) [inline src/app/app.ts:2]:
```suggestion
import {openConnection, closeConnection} from '../database/connectDB';
```

**Review by debasishbsws — CHANGES_REQUESTED** [review]

**Review by debasishbsws — APPROVED** [review]
Looks good.

**Pull request merged**